src/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.Unix.BackingObject_File.cs

```diff
@@ -37,6 +38,10 @@ private static FileStream CreateSharedBackingObject(Interop.libc.MemoryMappedPro
         // ---- PAL layer ends here ----
         // -----------------------------
 
-        private static readonly string TmpPathPrefix = Path.Combine(Path.GetTempPath(), MemoryMapObjectFilePrefix);
+        private static readonly string s_tempMapsDirectory = Path.Combine(
+            Path.GetTempPath(),
```

Comment:
Because System.IO.Pipes doesn't/can't depend on System.IO.FileSystem and thus can't use Directory.CreateDirectory, which supports creating all of the non-existing directories in the path.  That means for the would-be primary consumer of the API (currently), we'd be building up the path, only to then force System.IO.Pipes to need to parse that string in order to create each individual directory in the path.